docs/sources/v0.51.x/using-k6-browser/migrating-to-k6-v0-52.md

```diff
@@ -0,0 +1,229 @@
+import { browser } from 'k6/experimental/browser';
+
+export const options = {
+  scenarios: {
+    ui: {
+      executor: 'shared-iterations',
+      options: {
+        browser: {
+          type: 'chromium',
+        },
+      },
+    },
+  },
+  thresholds: {
+    checks: ['rate==1.0'],
+  },
+};
+
+export default async function () {
+  const context = await browser.newContext();
+  const page = await context.newPage();
+
+  try {
+    // Navigate to the website under test.
+    await page.goto('https://test.k6.io/', { waitUntil: 'networkidle' });
+    await Promise.all([
+      page.waitForNavigation(),
+      page.locator('a[href="/my_messages.php"]').click(),
+    ]);
+
+    // Check that the header reads "Unauthorized".
+    const h2 = page.locator('h2');
+    const headerOK = (await h2.textContent()) == 'Unauthorized';
+    check(headerOK, { header: headerOK });
+
+    // Take a screenshot of the page.
+    await page.screenshot({ path: 'screenshot.png' });
+  } finally {
+    await page.close();
+  }
+}
+```
+
+{{< /code >}}
+
+### Can I promise chain in v0.51?
+
+If you want to promise chain the newly migrated async APIs, You won't be able to do that until after the migration and release of k6 `v0.52`. For example, you can't do this yet:
+
+{{< code >}}
+
+```javascript
+import { browser } from 'k6/experimental/browser';
+
+// options block
+
+export default async function () {
+  const page = browser.newPage();
+
+  // ...
+
+  await page.$('#text1').then((text) => text.type('hello world'));
+
+  // ...
+}
+```
+
+{{< /code >}}
+
+You could do this instead:
+
+{{< code >}}
+
+```javascript
+import { browser } from 'k6/experimental/browser';
+
+// options block
+
+export default async function () {
+  const page = browser.newPage();
+
+  // ...
+
+  const text = await page.$('#text1');
+  await text.type('hello world');
+
+  // ...
+}
+```
+
+{{< /code >}}
+
+## Working with k6 check
+
+The k6 `check` API wasn't designed to work with async APIs. However there is a workaround. In your current scripts, you may have used a `check` like so:
+
+{{< code >}}
+
+```javascript
+import { browser } from 'k6/experimental/browser';
+import { check } from 'k6';
+
+// options block
+
+export default async function () {
+  const page = browser.newPage();
+
+  // ...
+
+  check(page, {
+    header: (p) => p.locator('h2').textContent() == 'Welcome, admin!',
+  });
+
+  // ...
+}
+```
+
+{{< /code >}}
+
+Since the `locator.textContent` API is being migrated to async, and `check` doesn't work with async APIs, you will need to do something like this:
+
+{{< code >}}
+
+```javascript
+import { browser } from 'k6/experimental/browser';
+import { check } from 'k6';
+
+// options block
+
+export default async function () {
+  const page = browser.newPage();
+
+  // ...
+
+  const h2 = page.locator('h2');
+  const headerOK = (await h2.textContent()) == 'Welcome, admin!';
+  check(headerOK, { header: headerOK });
+
+  // ...
+}
+```
+
+{{< /code >}}
```

Comment:
I like it. The issue is with the linter, it complains that check, page etc aren't defined in the example. I've amended your example a tiny bit to avoid the linter issue. Updated in 38cf3a175ddd13f9dd7bacbdf4a798908fc945ba.

@heitortsergent is there a way to bypass the javascript linter?

Reply:
I found a way to skip the linter.